addons/spreadsheet_account/models/account.py

```diff
@@ -79,19 +72,182 @@ def _build_spreadsheet_formula_domain(self, formula_params):
+
+        all_lines = {}
+        aggregates = [field + ':sum' for field in fields]
+
+        include_initial_balance_account_ids = []
+        for company_id, acc in accounts.items():
+            include_initial_balance_account_ids += acc.with_company(company_id).filtered('include_initial_balance').ids
+
+        for period_num, period in enumerate(timeline):
+            if period_num == 0:
+                company_ids = list({args['company_id'] or self.env.company.id for args in args_list})
+                account_ids = include_initial_balance_account_ids
+            else:
+                args_list_in_period = [args for args in args_list if period in args['date_periods']]
+                company_ids = list({args['company_id'] or self.env.company.id for args in args_list_in_period})
+                subcodes = tuple({subcode for args in args_list_in_period for subcode in args['codes'] if subcode})
+                account_ids_in_period = self.env['account.account']
+                for company_id in company_ids:
+                    account_ids_in_period |= accounts[company_id].filtered(
+                        lambda account:
+                        account.with_company(company_id).code.startswith(subcodes)
+                        or account.with_company(company_id).include_initial_balance
+                    )
+                account_ids = account_ids_in_period.ids
+
+            if not account_ids:
+                continue
+
+            domain = [
+                *self.env['account.move.line']._check_company_domain(company_ids),
+                ('account_id', 'in', account_ids),
+                ('parent_state', 'in', ('draft', 'posted')),
+                ('date', '>=', period[0]),
+                ('date', '<=', period[1]),
+            ]
+
+            lines_in_period = self.env['account.move.line'].with_context(allowed_company_ids=company_ids)._read_group(
+                domain=domain,
+                groupby=['company_id', 'parent_state', 'account_id', 'partner_id'],
+                aggregates=aggregates,
+            )
+            all_lines.update({
+                (period, company.id, state, account.id, partner.id): dict(zip(fields, val))
+                for company, state, account, partner, *val in lines_in_period
+            })
+        return all_lines
+
+    # ------------------------ #
+    #      API functions       #
+    # ------------------------ #
+
     @api.model
     def spreadsheet_move_line_action(self, args):
+        self._pre_process_date_period_boundaries([args])
         domain = self._build_spreadsheet_formula_domain(args)
+        codes = [code for code in args["codes"] if code]
+        partner_ids = [partner_id for partner_id in args.get('partner_id', []) if partner_id]
+        if codes:
+            name = _("Journal items for account prefix %s", ", ".join(codes))
+        elif partner_ids:
+            name = _("Journal items for partner(s) %s", ", ".join(partner_ids))
```

Comment:
Partner ids as integers?! :sweat_smile: 

Can't we just always do "Cell Audit" or something like that? That conveys that it is a domain limited view, without having 10km long titles (that would probably be cropped anyway)